tools/clang/lib/Parse/ParseExpr.cpp

```diff
@@ -454,16 +454,18 @@ Parser::ParseRHSOfBinaryExpression(ExprResult LHS, prec::Level MinPrec) {
 
         LHS = Actions.ActOnBinOp(getCurScope(), OpToken.getLocation(),
                                  OpToken.getKind(), LHS.get(), RHS.get());
-      } else
+      } else {
         LHS = Actions.ActOnConditionalOp(OpToken.getLocation(), ColonLoc,
                                          LHS.get(), TernaryMiddle.get(),
                                          RHS.get());
-    } else
+      }
+    } else {
+      // HLSL Change Begin - Take care TernaryMiddle.
```

Comment:
I believe this comment belongs after the else block, before:
```cpp
    // HLSL Change Begin - Take care TernaryMiddle.
    Actions.CorrectDelayedTyposInExpr(TernaryMiddle);
```

Note: The original bracing style was trying to preserve the existing code, unchanged, outside the `// HLSL Change Begin`/`End` marks.  Since we are unlikely to ever integrate upstream changes back into this codebase on mass, this matters less now.